symintify nll loss fns (#86915)

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87095

This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 

Follow up issue for other unhandled parameter types: https://github.com/pytorch/pytorch/issues/87283

anjali411 committed:
symintify nll loss fns (#86915)

This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

[ghstack-poisoned]

anjali411 commented on test/test_autocast.py:
```diff
@@ -116,7 +116,7 @@ def test_autocast_torch_fp32(self):
         for op_with_args in self.autocast_lists.nn_fp32:
             op, args, maybe_kwargs = self.args_maybe_kwargs(op_with_args)
-            self._run_autocast_outofplace(op, args, torch.float32, module=torch._C._nn, add_kwargs=maybe_kwargs)
+            self._run_autocast_outofplace(op, args, torch.float32, module=torch.ops.aten, add_kwargs=maybe_kwargs)
```
there's probably some python binding issue because calling into `torch._C._nn` leads to:
```
======================================================================
ERROR: test_autocast_nn_fp32 (__main__.TestAutocastCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch/chourdiaanjali/work/pytorch/test/test_autocast.py", line 120, in test_autocast_nn_fp32
    self._run_autocast_outofplace(op, args, torch.float32, module=torch._C._nn, add_kwargs=maybe_kwargs)
  File "/scratch/chourdiaanjali/work/pytorch/test/test_autocast.py", line 39, in _run_autocast_outofplace
    output = getattr(module, op)(*args, **add_kwargs)
RuntimeError: SymInt constructor with input: -4771225742975334011

----------------------------------------------------------------------
Ran 1 test in 1.805s

FAILED (errors=1)
```

ezyang replied:
ok, if you want to land this as a workaround it's good by me. I guess I'll look into the root cause then

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915

[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915

[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915

[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915

[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 


[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 


[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 


[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 


[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 


[ghstack-poisoned]

albanD commented on torch/csrc/utils/python_arg_parser.cpp:
```diff
@@ -817,17 +817,14 @@ auto FunctionParameter::check(
       return THPStream_Check(obj);
     case ParameterType::STRING:
       return THPUtils_checkString(obj);
-    default:
```
Woops

albanD reviewed (approved):
Sounds good.
Could you open a high pri issue to fix the missing types?

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 

Follow up issue for other unhandled parameter types: https://github.com/pytorch/pytorch/issues/87283


[ghstack-poisoned]

anjali411 committed:
Update on "symintify nll loss fns (#86915)"


This reverts commit bbd7b38d5580c44ffb4404d431e07bc2316e59d5.

Reland https://github.com/pytorch/pytorch/pull/86915 with a fix for python arg parser handing for SymInt and SymIntList. 
This was uncovered because we are calling directly into python bindings code through test_autocast.py (`torch._C._nn.nll_loss`)  without providing a value for the optional symint arg (`ignore_index`). The arg parser constructs the  SymInt and SymIntList using the recorded "default_int" or "default_int_list" (schema string parsing) in case a value is not received for an optional argument. Since we weren't handling the symint case properly, the default_int just had a garbage value which was later being used to construct SymInt. 

Follow up issue for other unhandled parameter types: https://github.com/pytorch/pytorch/issues/87283


[ghstack-poisoned]

anjali411 commented:
@pytorchbot merge -g

anjali411 commented:
@pytorchbot merge -f "unrelated test failures"
